src/main/java/com/money_account_service/services/AuthorizationService.java

```diff
@@ -1,22 +1,48 @@
 package com.money_account_service.services;
 
+import com.money_account_service.dtos.response.AuthorizeResponseDto;
+import com.money_account_service.entities.AccountEntity;
 import com.money_account_service.mappers.UserMapper;
 import com.money_account_service.models.UserModel;
+import com.money_account_service.repositories.AccountRepository;
 import com.money_account_service.utility.UserServiceClient;
 import lombok.AllArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.server.ResponseStatusException;
 
+import java.util.Optional;
+
 @AllArgsConstructor
 public class AuthorizationService {
 
     UserServiceClient userServiceClient;
 
-    public UserModel authorize(String token) {
+    AccountRepository accountRepository;
+
+    public Long authorize(String token) {
+
+        if (token.startsWith("Bearer")) {
+            return getAccountIdByUserSub(userServiceClient.authorize(token.substring(7)));
+        }
+        throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Wrong token provided");
+    }
+
+    public UserModel authorizeWithUserModel(String token) {
 
         if (token.startsWith("Bearer")) {
             return UserMapper.authorizeRequestDtoToUserModel(userServiceClient.authorize(token.substring(7)));
```

Comment:
Why mapper and not a regular constructor? Mappers mostly reside in-between layers. Here, it doesn't seem necessary.

Reply:
so we want to use just authorizeRequestDto?

Reply:
I meant I would write something like:
```
val userSub = userServiceClient.authorize(token.substring(7));
return UserModel(userSub)
```